sycl/source/detail/graph_impl.cpp

```diff
@@ -117,12 +74,26 @@ void duplicateNode(const std::shared_ptr<node_impl> Node,
   }
 }
 
+/// Recursively add nodes to execution stack.
+/// @param NodeImpl Node to schedule.
+/// @param Schedule Execution ordering to add node to.
+void sortTopological(std::shared_ptr<node_impl> NodeImpl,
+                     std::list<std::shared_ptr<node_impl>> &Schedule) {
+  for (auto &Succ : NodeImpl->MSuccessors) {
+    // Check if we've already scheduled this node
+    auto NextNode = Succ.lock();
+    if (std::find(Schedule.begin(), Schedule.end(), NextNode) == Schedule.end())
+      sortTopological(NextNode, Schedule);
```

Comment:
Stylistic nitpick, but think we use braces for single statement `if`s elsewhere
```suggestion
    if (std::find(Schedule.begin(), Schedule.end(), NextNode) == Schedule.end()) {
      sortTopological(NextNode, Schedule);
    }
```